pkg/solana/codec/time.go

```diff
@@ -0,0 +1,54 @@
+package codec
```

Comment:
nit: File name should be duration.  I don't recall, were you going to add time.Time later?  I thought you did the work to get it working in common.

Also, do you need set or map primitives that rust provides?

Reply:
I added a hook for `time.Time`. The default for `unixTimestamp` is int64 and I added a hook to allow mappings between the two.

The IDL does not include set or map primitives as far as I could tell, which was strange. The issue is that when parsing the IDL, if it doesn't have those structures available, I don't have a starting point in building those types with the codec.